### ISSUE TITLE
Discussion: Default memory allocation

### ISSUE THREAD

**Issue by fuyufjh** [risingwavelabs/risingwave#7683]
Given the total memory, CN should be able to allocate it automatically. **This issue focuses on the policy part**, while https://github.com/risingwavelabs/rfcs/pull/26 discussed the implementation part.

These numbers below are arbitrary. Please comment with your preferences and suggestions.

## Compute Node

- **Total** `total_memory_bytes`  
    - **Reserved** = min(total * 10%, 500MB)
    - Non-Reserved = total - reserved
        - **Batch** = 10%
        - **Streaming** = 60%
        - **Storage** = 30%
            - block_cache_capacity_mb
            - meta_cache_capacity_mb
            - compactor_memory_limit_mb
            - shared_buffer_capacity_mb

## Compactor

- Total Available Memory `total_memory_bytes`
    - Reserved = min(total * 10%, 500MB)
    - Non-Reserved = total - reserved
        - shared_buffer_capacity_mb

### Flink

References:

- [Apache Flink: Memory Management Improvements with Apache Flink 1.10](https://flink.apache.org/news/2020/04/21/memory-management-improvements-flink-1.10.html)
- [Apache Flink 1.10 Documentation: Set up Task Executor Memory](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/memory/mem_setup/)

In Short, their policy is

- Total Process Memory
  - JVM Overhead = max(min(total * 0.1, 192MB), 1GB)
  - Total Flink Memory = total - jvm_overhead
       - Managed Memory = 25%
         - Streaming state (RocksDB)
         - Batch spilling 
       - Network Buffers = 12.5%
       - Fixed component = 256MB
       - Task Heap = the rest
